### ISSUE TITLE
problem with dynamic cast to a generic type

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#6051]
The following example does not compile:

``` chapel
class Parent {
  var x:int;
}

class Child : Parent {
  var y: int;
}

class Grandchild : Child {
  var z: int;
  param p;
}

config param n = 10;
var A:[1..n] Parent;
for param i in 1..n {
  A[i] = new Grandchild(p=i, x=i, y=i, z=i);
}

var first:Parent = A[1];
writeln("Attempting dynamic cast to Child");
var c = first:Child;
writeln(c);
writeln("Attempting dynamic cast to Grandchild");
var gc = first:Grandchild(1);
writeln(gc);
```

**Comment by daviditen** [comment]
This appears to have been fixed between versions 1.15.0 and 1.16.0.

**Comment by mppf** [comment]
PR  #7703 will add this test. Thanks for noticing that it's been fixed, @daviditen !
